### PR TITLE
Analyze sample code

### PR DESCRIPTION
**Pull request by Hixie**
cc @abarth

### CODE REVIEW

File: dev/bots/analyze-sample-code.dart [hunk 1]
```diff
@@ -0,0 +1,262 @@
+import 'dart:async';
```
Comment (abarth) [inline dev/bots/analyze-sample-code.dart]: Copyright block

File: dev/bots/analyze-sample-code.dart [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+final String _flutterRoot = path.dirname(path.dirname(path.dirname(path.fromUri(Platform.script))));
+final String _flutter = path.join(_flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter');
+
+class Line {
+  const Line(this.filename, this.line, this.indent);
+  final String filename;
+  final int line;
+  final int indent;
+  Line get next => this + 1;
+  Line operator + (int count) {
```
Comment (abarth) [inline dev/bots/analyze-sample-code.dart]: Don't we write `operator+` ?

File: packages/flutter/lib/src/material/colors.dart [hunk 3]
```diff
@@ -368,10 +368,10 @@ class Colors {
   /// ## Sample code
   ///
   /// ```dart
-  ///  new Icon(
-  ///    icon: Icons.widgets,
-  ///    color: Colors.red[400],
-  ///  ),
+  /// new Icon(
+  ///   Icons.widgets,
```
Comment (abarth) [inline packages/flutter/lib/src/material/colors.dart]: 😄

File: packages/flutter/lib/src/widgets/framework.dart [hunk 4]
```diff
@@ -17,6 +17,15 @@ export 'package:flutter/foundation.dart' show FlutterError, debugPrint, debugPri
 export 'package:flutter/foundation.dart' show VoidCallback, ValueChanged, ValueGetter, ValueSetter;
 export 'package:flutter/rendering.dart' show RenderObject, RenderBox, debugDumpRenderTree, debugDumpLayerTree;
 
+// Examples can assume:
+// BuildContext context;
+// void setState(VoidCallback fn) { }
+
+// Examples can assume:
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/framework.dart]: Why two blocks?

**Comment by abarth** [comment]
Gold star

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/colors.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/colors.dart]: Indeed!

File: packages/flutter/lib/src/widgets/framework.dart (hunk 4 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: The second block is classes that I'm hoping to eventually turn into actual sample code somewhere.

File: dev/bots/analyze-sample-code.dart (hunk 2 above)
Comment (Hixie) [inline dev/bots/analyze-sample-code.dart]: Apparently we write `operator +(...)`

File: dev/bots/analyze-sample-code.dart (hunk 1 above)
Comment (Hixie) [inline dev/bots/analyze-sample-code.dart]: Fixed. Also added to test.dart, which I started this one from...

File: dev/bots/analyze-sample-code.dart (hunk 2 above)
Comment (abarth) [inline dev/bots/analyze-sample-code.dart]: ok

**Pull request merged**